lib/SILOptimizer/PassManager/PassPipeline.cpp

```diff
@@ -891,6 +891,13 @@ SILPassPipelinePlan::getIRGenPreparePassPipeline(const SILOptions &Options) {
   // boundaries as required by the ABI.
   P.addLoadableByAddress();
 
+  if (Options.EmbeddedSwift) {
```

Comment:
Is this pass run really needed? There is already a LateDeadFunctionElimination earlier.

Reply:
I think it's only present in the -O pipeline. But I think we need it to happen in -Onone too, and the embedded/debuginfo.swift shows a problem specifically about -Onone + -g. Any better suggestions for how to ensure this pass runs in all pipelines?

Reply:
you could add it to `getOnonePassPipeline`

Reply:
Moved.